tests/testthat/test-mwrite.R

```diff
@@ -0,0 +1,207 @@
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with mrgsolve.  If not, see <http://www.gnu.org/licenses/>.
+
+library(testthat)
+library(mrgsolve)
+
+context("test-mwrite")
+
+test_that("convert list to code", {
+  # This could be simpler, but haven't found an easy way to do `c = "a"`
+  l <- list(a = 1, b = 1.234, c = "a", d = c("x", "y", "z"), e = FALSE)  
+  x <- mrgsolve:::tocode(l)  
+  expect_equal(x[1], "a = 1")
+  expect_equal(x[2], "b = 1.234")
+  expect_equal(x[3], 'c = "a"')
+  expect_equal(x[4], 'd = c("x", "y", "z")')
+  expect_equal(x[5], "e = FALSE")
+})
+
+test_that("convert model to list", {
+  mod <- house(end = 26, delta = 2, outvars = "GUT, DV, CP")
+  l <- mrgsolve:::mwrite_model_to_list(mod)
+  expect_equal(l$format, "list")
+  expect_equal(l$version, 1)
+  expect_equal(l$update$end, 26)
+  expect_equal(l$update$delta, 2)
+  expect_equal(l$update$outvars, c("GUT", "DV", "CP"))
+  expect_equal(l$set$end, 120)
+  expect_equal(l$set$delta, 0.25)
+  
+  nomega <- nrow(omat(mod))
+  expect_length(l$omega, 3)
+  expect_equal(names(l$sigma), c("data", "labels", "names"))
+  expect_is(l$omega$data, "list")
+  expect_length(l$omega$data, 1)
+  expect_length(l$omega$data[[1]], nomega)
+  expect_length(l$omega$labels, 1)
+  expect_length(l$omega$labels[[1]],  nomega)
+  expect_is(l$omega$labels, "list")
+  expect_length(l$omega$names, 1)
+  expect_is(l$omega$names, "character")
+  
+  mod2 <- mod
+  mod2@omega <- omat()
+  l <- mrgsolve:::mwrite_model_to_list(mod2)
+  expect_length(l$omega, 3)
+  expect_equal(names(l$sigma), c("data", "labels", "names"))
+  expect_is(l$omega$data, "list")
+  expect_length(l$omega$data, 0)
+  expect_length(l$omega$labels, 0)
+  expect_is(l$omega$labels, "list")
+  expect_length(l$omega$names, 0)
+  expect_is(l$omega$names, "character")
+})
+
+test_that("mwrite, mread yaml", {
+  skip_if_not_installed("yaml")
+  temp <- tempfile()
+  mod <- modlib("pk1", compile = FALSE)
+  mod <- update(mod, end = 16, delta = 8)
+  mwrite_yaml(mod, temp)
+  expect_true(file.exists(temp))
+  yaml <- readLines(temp)
+  l <- yaml::yaml.load(yaml)
+  expect_equal(l$format, "yaml")
+  expect_equal(l$version, 1)
+  expect_equal(l$param$V, 20)
+  
+  # Read back in 
+  mod2 <- mread_yaml(temp, compile = FALSE)
+  expect_identical(param(mod), param(mod2))
+  expect_identical(stime(mod), stime(mod2))
+  
+  # Switch up the model name
+  mod3 <- mread_yaml(temp, model = "foo", compile = FALSE)
+  expect_equal(mod3@model, "foo_mod")
+  
+  # We can pass capture through
+  mod4 <- mread_yaml(
+    temp, compile = FALSE, capture = "KA"
+  )
+  expect_true("KA" %in% outvars(mod4)$capture)
+})
+
+test_that("yaml_to_cpp", {
+  mod <- modlib("popex", compile = FALSE)
+  temp <- tempfile()
+  mwrite_yaml(mod, temp)
```

Comment:
This spot (line 102) will fail if the yaml package isn't installed, so this `testthat` block should call `skip_if_not_installed("yaml")` like other spots in the tests.

Here are additional `mwrite_yaml` calls in the tests that aren't guarded:

-   [tests/testthat/test-mwrite.R:119](https://github.com/metrumresearchgroup/mrgsolve/blob/aa94b84cbaaa5efba9e385b9f6b2897e70509f20/tests/testthat/test-mwrite.R#L119)
-   [tests/testthat/test-mwrite.R:132](https://github.com/metrumresearchgroup/mrgsolve/blob/aa94b84cbaaa5efba9e385b9f6b2897e70509f20/tests/testthat/test-mwrite.R#L132)
-   [tests/testthat/test-mwrite.R:144](https://github.com/metrumresearchgroup/mrgsolve/blob/aa94b84cbaaa5efba9e385b9f6b2897e70509f20/tests/testthat/test-mwrite.R#L144)

Reply:
Thanks; fixed these.